butterknife-runtime/src/test/java/butterknife/BindViewTest.java

```diff
@@ -49,12 +57,21 @@
         + "}"
     );
 
+    ButterKnifeProcessor butterKnifeProcessor = new ButterKnifeProcessor();
     assertAbout(javaSource()).that(source)
         .withCompilerOptions("-Xlint:-processing", "-Abutterknife.debuggable=false")
-        .processedWith(new ButterKnifeProcessor())
+        .processedWith(butterKnifeProcessor)
         .compilesWithoutWarnings()
         .and()
         .generatesSources(bindingSource);
+
+    Map<String, List<Element>> map = butterKnifeProcessor.getMapGeneratedFileToOriginatingElements();
+    assertEquals(1, map.size());
+    Map.Entry<String, List<Element>> entry = map.entrySet().iterator().next();
+    assertEquals("test/Test_ViewBinding.java", entry.getKey());
+    List<Element> elements = entry.getValue();
+    assertEquals(1, elements.size());
+    assertEquals("test.Test", elements.get(0).asType().toString());
```

Comment:
In this test, I use the method and the map I have added to the BK processor. 

There are different options here:
1) unit testing: we can test that we pass the right originating element to the filer. This is a healthy practice, idependently of gradle's incremental solution. 
2) integration testing. I've seen other projects using integration tests with gradle, with fixtures, and I am not very familiar with this technology. 

I decided to go with the first testing strategy as it limits the impact of the changes here, I also think this test and the tested behavior is just "healthier" for the compiler. But we can also prefer to test more directly that incremental annotation processing is working as expected, which would be better to ensure there is no regression at this level. It would probably bring more value for maintenance. 
If we decide to go this way though, I will need some help to write such a test.